src/hacl.rs

```diff
@@ -1,7 +1,62 @@
-//! # HACL
+//! # Low-level HACL APIs
 //!
-//! Interface to call into HACL functions.
+//! Unsafe API for HACL.
+//!
+//! |             | x86 | x86-64             | Arm32 | Arm64 | s390x |
+//! | ----------- | --- | ------------------ | ----- | ----- | ----- |
+//! | Portable C  | ✓   | ✓                  | ✓     | ✓     | ✓     |
+//! | simd128     | -   | SSE2, SSE3, SSE4.1 | -     | NEON  | z14   |
+//! | simd256     | -   | AVX, AVX2          | -     | -     | -     |
+
+#[cfg(aes_ni)]
+pub(crate) mod aesgcm;
+pub(crate) mod blake2;
+pub(crate) mod chacha20_poly1305;
+pub(crate) mod curve25519;
+pub(crate) mod drbg;
+pub(crate) mod ed25519;
+pub(crate) mod hkdf;
+pub(crate) mod hmac;
+pub(crate) mod p256;
+pub(crate) mod sha2;
+pub(crate) mod sha3;
+
+/// Unified error type.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Error {
```

Comment:
Do we need this error type or shall would it make more sense to pull this up into the main error type of libcrux?
This was necessary in the `hazmat` case. But I think it might be nicer if we skip it here.

Reply:
This `enum` is used in `signature.rs` and `ecdh`. I don't see where the main error type of libcrux.

Reply:
#65